packages/apollo-language-server/src/project/client.ts

```diff
@@ -211,10 +219,109 @@ export class GraphQLClientProject extends GraphQLProject {
   get clientSchema(): DocumentNode {
     return {
       kind: Kind.DOCUMENT,
-      definitions: this.typeSystemDefinitionsAndExtensions
+      definitions: [
+        ...this.defaultClientDefinitions,
+        ...this.typeSystemDefinitionsAndExtensions
+      ]
     };
   }
 
+  get defaultClientDefinitions(): readonly DefinitionNode[] {
+    return new GraphQLDocument(
+      new Source(`
+      """
+      Direct the client to resolve this field locally, either from the cache or local resolvers.
+      """
+      directive @client(
+        """
+        When true, the client will never use the cache for this value. See
+        https://www.apollographql.com/docs/react/essentials/local-state/#forcing-resolvers-with-clientalways-true
+        """
+        always: Boolean
+      ) on FIELD
+
+      """
+      Export this locally resolved field as a variable to be used in the remainder of this query. See
+      https://www.apollographql.com/docs/react/essentials/local-state/#using-client-fields-as-variables
+      """
+      directive @export(
+        """
+        The variable name to export this field as.
+        """
+        as: String!
+      ) on FIELD
+
+      """
+      Specify a custom store key for this result. See
+      https://www.apollographql.com/docs/react/advanced/caching/#the-connection-directive
+      """
+      directive @connection(
```

Comment:
Currently this seems to equal Relay’s version of the directive, but I don’t know if that’s a guarantee and so perhaps this could yield wrong information to a Relay user?

Reply:
Actually I was wrong, this doesn’t match Relay’s directive, which has a `filters` parameter (plural) https://relay.dev/docs/en/pagination-container#connection

Reply:
This is a great example of why the Apollo directives should be overridable.